Jenkinsfile

```diff
@@ -16,8 +16,14 @@ pipeline {
      		           
                 parallel(
                     "Print Environment": { sh 'env' },
-                    "Runtime Tests": { sh './contrib/vagrant/start.sh' },
-                    "K8s multi node Tests": { sh './tests/k8s/start.sh' }
+                    "Runtime Tests": {
+                         // Make sure that VMs from prior runs are cleaned up in case something went wrong in a prior build.
+                         sh 'vagrant destroy -f || true'
+                         sh './contrib/vagrant/start.sh'
+                     },
+                    "K8s multi node Tests": {
+                         sh './tests/k8s/start.sh'
```

Comment:
There is no risk of such an issue occurring. Vagrant works by finding the closest Vagrantfile that it can reach and ascends up in directories until it finds one. Check [here](https://www.vagrantup.com/docs/vagrantfile/) for more information, specifically:

> When you run any vagrant command, Vagrant climbs up the directory tree looking for the first Vagrantfile it can find, starting first in the current directory. So if you run vagrant in /home/mitchellh/projects/foo, it will search the following paths in order for a Vagrantfile, until it finds one: